tests/ut/ir/transforms/test_lower_composite_ops.py

```
@@ -455,5 +457,192 @@ def test_cos_in_return_stmt_is_decomposed():
+
+    * Phase 2a (notify all peers) — for + if
+    * Phase 2b (wait on all peers) — for + if
+    * Phase 3  (reduce-load from all peers) — for + if
+    * Phase 3.5a (post-reduce re-notify) — for + if
+    * Phase 3.5b (post-reduce re-wait) — for + if
+
+    Phase 3.5 is a second cross-rank barrier inserted between Phase 3
+    (read peers via ``pld.tile.remote_load``) and Phase 4 (write reduced
+    value back into ``target``). Without it, a fast rank could overwrite
+    its slot while slower ranks are still reading the staged Phase-1 data
+    — a write-after-read race that manifests as off-by-N×peer drift on
```

Ruff (0.15.15)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)